raiden/tests/integration/long_running/test_integration_events.py

```diff
@@ -557,8 +558,7 @@ def has_initiator_events():
     # A ChannelClose event will be generated, this will be polled by both apps
     RaidenAPI(app0.raiden).channel_close(registry_address, token_address, app1.raiden.address)
 
-    exception = ValueError("Could not get close event")
-    with gevent.Timeout(seconds=30, exception=exception):
+    with block_offset_timeout(app0.raiden, "Could not get close event"):
```

Comment:
Note for transaction timeouts: Both approaches here are a little arbitrary, the previous code had an arbitrary value of 30 seconds, this context manager has an arbitrary block timeout of `settle_timeout`. Even though the values are arbitrary, the version based on the block height can account for variations on mining times, so it should be slightly better

Reply:
We have a lot more 10-second or 30-second timeouts, some depending on the blockchain type or using `estimate_blocktime`. It might make sense to do a follow-up and change more of the timeouts in the integration tests with x blocks instead of x seconds.